### DISCUSSION TITLE
Micromagnetics in Phase Field

### DISCUSSION THREAD

**Discussion by alanchavez86** [idaholab/moose#17074] [Q&A General] [answered]
Hi all!

I was wondering if micromagnetics have been implemented in any phase field models and was curious if anyone is interesting in sharing how they implemented it in their models. 

I'm looking into incorporating demagnetization energy (E_d) from dipole-dipole interactions which influences the behavior of the phase decomposition of FeCrCo. 

I would like to know if there is a simpler way to implement equation 6 from Toshiyuki Koyama 2008 Sci. Technol. Adv. Mater. 9 013006 (paper attached). 

Thank you all for the help.

Alan C.
[Koyama_2004.pdf](https://github.com/idaholab/moose/files/6007199/Koyama_2004.pdf)

**Comment by mangerij** [comment, marked as answer]
Hi Alan,

Micromagnetics is implemented in Ferret: https://mangerij.github.io/ferret/
We use a Landau-Lifshitz-Bloch which handles finite temperature fluctuations and is used to constrain the magnetization vector to a fixed length. At 0K it reduces to the LLG problem. We have tested exchange, demag, and anisotropy couplings and all have reasonable agreement to other micromagnetic codes (i.e. muMax3 and Vampire).

Concerning the 2004 Koyama paper: we have been approached already regarding this paper and the tentative conclusion is that their model is a bit incomplete and LLG is unnecessary. 

1.) The phase transitions you are investigating are at very high temperatures, therefore fluctuations of the magnetic order parameter would probably kill any domain ordering/magnetic precession you would get from solving the LLG equation. 

2.) The timescales on magnetic relaxation/precession are much much shorter than the spinodal decomposition process. Therefore, evolving the LLG+CH problem would take a very long time. The exchange length also is much much smaller than the CH interfacial width posing a problem to resolve magnetic topology.

3.) The magnetic field H = -grad(Φ) can be calculated by just solving the Poisson equation with Dirichlet conditions on the opposing boundaries to generate an applied field. In the Poisson equation, you just include coupling to your concentration given magnetization saturation dependence on the concentration. This is possible and should be fairly straightforward. You are welcome to look at Ferret on how the Poisson equation is solved (but it is trivially a diffusion equation for Φ and addition of a kernel that just does divM). But since you have M = M[c(r)], you need div(M[c(r)] which is a nasty expression. I would suggest using the AD system for this. 

So my point is that the driving force for the 2D elongation of the phases along the field is because he includes a Zeeman term and also couples to the concentration via the demag field resulting from the Poisson equation and not from the other terms (exchange, anisotropy, magnetostriction, etc) from a "micromagnetic" model. I am speculating that you might get the same results ignoring demag completely and just including the Zeeman term in your CH equation - this might be a wise place to start.

In his later paper (see Koyama 2008 Sci. Technol. Adv. Mater. 9 013006), he includes an easier to follow equation set by the way.
Cant figure how to attach here...

John

**Reply by alanchavez86** [reply]
John.

Thank you for the insight and information. I'll look into Koyama's later paper and try to follow that equation as well as working with Ferret. My mentor also provided me a document regarding Micromagnetics in Ferret so I'll be looking through that as well.

Thank you for your help.

Alan
